### ISSUE TITLE
NW app not launching randomly on windows 7 32 bit machine.

### ISSUE THREAD

**Issue by anurag2911** [nwjs/nw.js#7037]
======== TEMPLATE BUG FORM ========

NWJS Version :nwjs-v0.25.4-win-ia32
Operating System :windows 7 32 bit

### Expected behavior
NW app not launching randomly on windows operating system startup. Our product ZENworks/ZAPP is built on nw.js and typically on OS startup we launch it and see 4 processes instance of nw.exe(which is renamed as zapp.exe for our case)in process manager but when this issue happens we see 3 processes instance.

All instances should be running so that we can display the UI to end user.

This issue is random and on a few reboots of machine happens once. I have captured a trace log which I am attaching here.

**Comment by anurag2911** [comment]
[zapptrace.log](https://github.com/nwjs/nw.js/files/3067987/zapptrace.log)

**Comment by ffanny** [comment]
I test [zapp](https://www.novell.com/communities/coolsolutions/cool_tools/zapp/) with 4 .exe files in ZfD toolkit files and ZENAPP32.DLL on my Windows 10. I didn't find 4 or 3 processes instance of nw.exe(which is renamed as zapp.exe) in process manager on OS startup.

Would you please provide a photo of the processes in this case in process manager?  Or tell me if I do the wrong work to reproduce.

**Comment by anurag2911** [comment]
> Would you please provide a photo of the processes in this case in process manager? Or tell me if I do the wrong work for reproducing.
![zappprocesssnapshot](https://user-images.githubusercontent.com/17509728/55966626-5c021180-5c96-11e9-92e7-1ca04b796cb0.png)

**Comment by anurag2911** [comment]
nw.exe is renamed as zapp.exe in our case.I have attached snapshot also.

**Comment by anurag2911** [comment]
> I test [zapp](https://www.novell.com/communities/coolsolutions/cool_tools/zapp/) with 4 .exe files in ZfD toolkit files and ZENAPP32.DLL on my Windows 10. I didn't find 4 or 3 processes instance of nw.exe(which is renamed as zapp.exe) in process manager on OS startup.
> 
> Would you please provide a photo of the processes in this case in process manager? Or tell me if I do the wrong work for reproducing.

That's not the zapp I am talking about. That is something else which you have found in https://www.novell.com/communities/coolsolutions/cool_tools/zapp/

**Comment by ffanny** [comment]
Could you please give me a download method of your product ZENworks/ZAPP to help me reproducing?

**Comment by ffanny** [comment]
And  [the crash dump file](http://docs.nwjs.io/en/latest/For%20Developers/Understanding%20Crash%20Dump/#find-minidump-file) is useful and needed if you can find and provide it to me.

**Comment by anurag2911** [comment]
OK, Thanks for looking into this. I want to upload an MSI file but its size is around 54 MB and it is not allowed to upload >10 MB here. With that MSI file, you can install ZAPP and write some script to launch on windows log-on also by double-clicking you can launch. Please provide an upload location.
Regarding the crash dump, I will get from my customer and upload.

**Comment by anurag2911** [comment]
> Could you please give me a download method of your product ZENworks/ZAPP to help me reproducing?

Please provide a share where I can upload an MSI file using which you can install ZAPP product. I can't paste it here because of MSI size.

**Comment by anurag2911** [comment]
> And [the crash dump file](http://docs.nwjs.io/en/latest/For%20Developers/Understanding%20Crash%20Dump/#find-minidump-file) is useful and needed if you can find and provide it to me.

Actually, it is not crashing, typically we see 4 instances of nw.exe process running but in case of NW app not launching we see 3 instances running. Do you know any troubleshooting steps on how to get more logs or something to check till which point it is going?

**Comment by ffanny** [comment]
You can share the MSI file using [google drive](https://drive.google.com/drive/my-drive) or other sharing platforms.

**Comment by anurag2911** [comment]
I have uploaded msi in this location:
https://www.dropbox.com/s/25d6b2343m1s13i/novell-zenworks-application-19.0.0.2546.msi?dl=0

Please right click and install this MSI ,it will create a folder structure like \Novell\ZENworks\zapp.

You can find zapp.exe file present inside this folder ,please double click and you can see in task manager 4 instances of zapp.exe running.

To replicate customer scenario ,please launch this on windows logon.

**Comment by ffanny** [comment]
I can see 6 instances of zapp.exe in task manager when I double click it and the status is like the screenshot shows. 
![TIM图片20190422103352](https://user-images.githubusercontent.com/19678102/56479879-3244b800-64ea-11e9-8159-35f0264afe9f.png)

**Comment by anurag2911** [comment]
I think you have clicked twice maybe so you are seeing these many instances. My question is more around if nw app is stuck somewhere while launching how I can troubleshoot.
The issue customer is facing is random so I can't reproduce and show you, it happens during Windows start-up randomly. We set zapp.exe (which is actually nw.exe) to launch and windows user login.

**Comment by anurag2911** [comment]
Is there a way to troubleshoot such scenarios, like enabling more logging or anything else?

Our product which is nw.exe renamed as zapp.exe is not launching randomly on windows 7 32 bit machines.
Any help will be much appreciated

**Comment by rogerwang** [comment]
I think maybe one of the nw processes crashed. Please try to locate a crash dump file.

**Comment by anurag2911** [comment]
Yes, I asked my customer about the crash dump file but they could not find any in their system, so I feel it is getting stuck somewhere while starting. When it is fully up and running then our app will make some logging but it is not reaching to that point.

**Comment by rogerwang** [comment]
I suggest updating to latest NW.js stable. v0.25 is quite outdated.

**Comment by anurag2911** [comment]
> I suggest updating to latest NW.js stable. v0.25 is quite outdated.

This can be tried, but our product is tested on v0.25 completely so any change in version will involve testing all use cases again.

**Comment by anurag2911** [comment]
I built my application on nw.js version 0.39.2 and gave to the customer, he still sees the issue that randomly zapp.exe(nw.exe we renamed as zapp.exe) doesn't launch during Windows start-up.
On double click, it gets launched. When the issue happens then 2 instances of zapp.exe(nw.exe) is shown in task manager, otherwise for our js file usually we see 4 instances. No crash dump generated anywhere but just two instances of zapp.exe(nw.exe) in task manager. I have asked the customer to provide process dump of those 2 instances.

Any thoughts on how to resolve this issue or any hint what could be possibly going wrong.

**Comment by anurag2911** [comment]
I have process dumps with me taken from customer setup where not all nw.exe processes are launched(zapp.exe in our case) and as a result user doesn't see the user interface.
Can you please analyze them? It is urgent for this customer.

**Comment by anurag2911** [comment]
Any idea when you can look into this issue?

**Comment by rogerwang** [comment]
Could you provide the crash dump?

**Comment by anurag2911** [comment]
Please note that nw.exe is not crashing but it is not creating all the child processes needed to run our application.

**Comment by mrunzo** [comment]
I am also facing this problem but in my case my app don't run correctly, any idea?

**Comment by anurag2911** [comment]
With another customer: nw.exe processes are in kilobytes of memory randomly , usually, it comes in megabytes for our angular js application.

I am afraid with nw.exe now running in millions of customer devices and no support from the nw.js team for our issue will force us to move to the electron.

**Comment by rogerwang** [comment]
Please provide the crash dump described in https://nwjs.readthedocs.io/en/latest/For%20Developers/Understanding%20Crash%20Dump/#find-minidump-file
Not the windows DMP file.

**Comment by anurag2911** [comment]
Unfortunately, there is no crash dump generated. From DMP file we need to check where it is stuck.

**Comment by anurag2911** [comment]
@rogerwang, one more customer has seen this issue that nw.exe randomly is not launching during windows login.
We launch nw.exe during windows startup and typically 4-5 processes are created, whenever this issue is seen then 1-2 processes are seen in the task manager. On killing those processes and launching nw.exe manually works. Is there a way I can enable full logging for nw.exe? I don't think it is a crash , it is just stuck somewhere and all nw.exe processes are not launching.
